src/script.js

```diff
@@ -151,9 +151,9 @@ function decompilePushOnly (script) {
 
   return chunks.map(function (op) {
     if (Buffer.isBuffer(op)) return op
-    if (op === OPS.OP_0) return new Buffer(0)
+    if (op === OPS.OP_0) return op
 
-    return scriptNumber.encode(op - OP_INT_BASE)
+    return op
   })
```

Comment:
if you don't do this, you end up with something like: 
decompiled script == [OP_0, Buffer(sig1), Buffer(sig2), Buffer(redeemScript)

but the witness == ['', Buffer(sig1), Buffer(sig2), Buffer(redeemScript)]

which obviously aren't the same. The script stack is the common representation between the two though!